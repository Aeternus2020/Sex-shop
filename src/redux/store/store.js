import { applyMiddleware, combineReducers, createStore } from "redux";
import products from "../getProducts/reducers";
import basket from "../pages/basket/reducers";
import favorite from "../pages/favorite/reducers";
import showModal from "../components/modal/reducers"; 
import title from "../pages/app/reducers";
import { composeEnhancers, middleware } from "./middleware";

export const rootReducer = combineReducers({
    products,
    basket,
    favorite,
    showModal,
    title
})

export const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(...middleware))
    )