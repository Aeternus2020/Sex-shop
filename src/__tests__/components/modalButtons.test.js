import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import ButtonModal from "../../components/buttonModal";
import { Provider } from "react-redux";
import { store } from "../../redux/store/store";
import renderer from 'react-test-renderer';

afterEach(cleanup);

test("Test ModalButton", () => {
    const item = {
        name: "Bondage",
        price: "70$",
        url: "./images/bondage.webp",
        article: "25987456",
        color: "black",
    };

    render(
    <Provider store={store}>
        <ButtonModal id={3} item={item} />
    </Provider>
    );

    expect(screen.getByTestId("buttons-modal-3")).toBeInTheDocument();
    fireEvent.click(screen.getByRole("button"));
});

test('Test snapshot ButtonModal', () => {
    const tree = renderer.create(
    <Provider store={store} >
        <ButtonModal id={3}/>
    </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})