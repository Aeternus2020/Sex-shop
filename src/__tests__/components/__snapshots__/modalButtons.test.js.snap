// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot ButtonModal 1`] = `
<button
  className="undefined"
  data-testid="buttons-modal-3"
  onClick={[Function]}
/>
`;
